modules/public/views.py

```diff
@@ -162,24 +162,63 @@ def convert_limit_offset_to_size_page(link):
 
 
 def homepage():
+    featured_snaps = []
+    errors = []
+    try:
+        featured_snaps = normalize_searched_snaps(api.get_featured_snaps())
+    except api.InvalidResponseContent as invalid_response_content:
+        errors = ["Error from server"]
+    except api.ApiErrorResponse as api_error_exception:
+        if api_error_exception.errors:
+            errors = api_error_exception.errors
+        else:
+            errors = ["Unknown error"]
```

Comment:
I believe we're masking the error message from the API layer in case it leaks secrets of some kind? It's probably worth mentioning here at least that it's an error with the API's response. We should also at least [log](http://flask.pocoo.org/docs/dev/logging/) the original error message.

Reply:
I can add better errors descriptions, can we add the log once the refactor is done. Maybe as a new issue in the epic?